packages/dynamodb-emulator/deamon.js

```diff
@@ -0,0 +1,154 @@
+const path = require('path');
+const fs = require('fs');
+const http = require('http');
+const qs = require('querystring');
+const { launch } = require('./index');
+const log = require('logdown')('dynamodb-emulator:deamon');
+
+const pidPath = path.join(__dirname, '.pid');
+// eslint-disable-next-line
+
+const getHash = () => {
+  const content = fs.readFileSync(__filename);
+  return require('crypto')
+    .createHash('md5')
+    .update(content)
+    .digest('hex');
+};
+
+const launchedVersion = getHash();
+
+let globalEmulator = null;
+let resolvedGlobalEmulator = null;
+let server = null;
+
+async function getEmulator(opts) {
+  if (globalEmulator) {
+    return globalEmulator;
+  }
+
+  // if the emulator stops then this process should also close.
+  globalEmulator = launch(opts);
+  let emu;
+  try {
+    emu = await globalEmulator;
+  } catch (err) {
+    if (err) {
+      log.error(err);
+      server.close();
+    }
+  }
+  resolvedGlobalEmulator = emu;
+  emu.proc.once('exit', () => {
+    server.close();
+  });
+  return emu;
+}
```

Comment:
Shouldn't `globalEmulator` get assigned to resolvedGlobalEmulator?

Reply:
I intentionally kept the data types consistent so globalEmulator is always a promise (even if it's resolved) resolvedGlobalEmulator is sort of hacky but is there for graceful cleanup.